### ISSUE TITLE
Odd activity auto tracing duration

### ISSUE THREAD

**Issue by LouisFn** [getsentry/sentry-java#1787]
_Platform:_
- [X]  Android -> minSdk: 24, compileSdk: 30, targetSdk: 30, buildTools: 31.0.0
- [X]  Kotlin -> 1.5.31, jvmTarget: VERSION_11
- [X]  Timber -> 5.0.1

_Build system:_
- [X]  Gradle -> 7.2

_Android Gradle Plugin: 7.0.2

The version of the SDK:
**5.2.0**

---
I have the following issue:

We get odd result with activity auto tracing. We receive many transactions with a very long duration on prod. 

<img width="1204" alt="Screenshot 2021-10-29 at 10 59 40" src="https://user-images.githubusercontent.com/54979514/139409679-ff0c67ec-d63a-451e-a260-81e5120dfb0c.png">

**Comment by marandaneto** [comment]
@LouisYubo I can't really reproduce it, not sure how to proceed, any ideas on how to reproduce otherwise I'd have to close it.

**Comment by marandaneto** [comment]
@LouisFn does your app have notifications? is it possible that users open your app via notification?

**Comment by LouisFn** [comment]
Completely, I can reproduce it by opening the app from notification

**Comment by marandaneto** [comment]
@LouisFn ok thats a good start, can you tell me the exact steps to reproduce it? or provide a minimal reproducible example? thanks

**Comment by LouisFn** [comment]
![Screenshot 2021-11-16 at 14 12 16](https://user-images.githubusercontent.com/82385769/141992227-715b1689-4449-4eea-a7a2-b942a5d05610.png)

@marandaneto It seems to be because you are waiting for the child traces to finish before stoping the transaction (waitForChildren) and the child trace from FragmentLifecycleIntegration never ends.

It's due to NavDeepLinkBuilder, when the deep link is triggered, the destination and all of its parents will be on the back stack. But for its parents, only onCreate is called but not onResume. So the HomeFragment trace never finished.

You can find a reproducible example here:
https://github.com/LouisFn/SentryFrozenFrame 

BTW, I think it would be usefull to add an option to disable "waitForChildren"

**Comment by marandaneto** [comment]
will have a look, thanks

**Comment by marandaneto** [comment]
I can confirm that `HomeFragment` isn't calling `onFragmentResumed`

**Comment by marandaneto** [comment]
@LouisFn in your use case, maybe there's a bug in your example as well, you set:

`app:startDestination="@id/navigation_fragment_home"`

but in `LaunchActivity`, you set `.setDestination(R.id.navigation_fragment_home)`, so `HomeFragment` isn't even visible or anything, looks like a wrong setup of the `navigation-ui` rather than the bug in Sentry SDK.

**Comment by marandaneto** [comment]
do `setDestination` and `app:startDestination` must match or not? didn't find docs for that.

**Comment by LouisYubo** [comment]
> @LouisFn in your use case, maybe there's a bug in your example as well, you set:
> 
> `app:startDestination="@id/navigation_fragment_home"`
> 
> but in `LaunchActivity`, you set `.setDestination(R.id.navigation_fragment_home)`, so `HomeFragment` isn't even visible or anything, looks like a wrong setup of the `navigation-ui` rather than the bug in Sentry SDK.

I don't understand what you mean, I set `.setDestination(R.id.navigation_fragment_chat)` no `.setDestination(R.id.navigation_fragment_home)` so it's normal if HomeFragment isn't visible



> do `setDestination` and `app:startDestination` must match or not? didn't find docs for that, if they don't have to, maybe fixing on the SDK would make more sense.

No, the purpose is to navigate to a specific destination (not the start destination) while adding the start destination to the backstack. 

From the docs: https://developer.android.com/reference/androidx/navigation/NavDeepLinkBuilder

>  When this deep link is triggered:
> The task is cleared.
> The destination and all of its parents will be on the back stack.
> [...]
> The parent of the destination is the start destination of the containing navigation graph.

**Comment by marandaneto** [comment]
sorry it was a typo, `.setDestination(R.id.navigation_fragment_chat)` indeed.

> No, the purpose is to navigate to a specific destination (not the start destination) while adding the start destination to the backstack.

got it, that makes sense, since the fragment does not trigger any other lifecycle, I don't really know how to finish the span automatically for that case, ideas?

**Comment by marandaneto** [comment]
even if I provide a custom `waitForChildren`, how would you know if `HomeFragment` was just added to the back stack but never visible, programmatically?

**Comment by LouisYubo** [comment]
Maybe, you can check if fragment `isAdded` in onCreate before starting the trace ?

**Comment by marandaneto** [comment]
indeed @LouisYubo that's totally possible and it works, thanks for the tip, See PR https://github.com/getsentry/sentry-java/pull/1813

**Comment by LouisYubo** [comment]
Thanks a lot for the fix :) 

But I think there are others issues with the auto tracing activity that doesn't seem to be related to `waitForChildren`. For now I didn't managed to reproduce this case

![Screenshot 2021-11-18 at 13 50 56](https://user-images.githubusercontent.com/82385769/142418611-e9c24305-50df-430f-bf47-b2277d2f7873.png)

or 

![Screenshot 2021-11-18 at 13 52 56](https://user-images.githubusercontent.com/82385769/142419093-4f6e6fb5-487e-403a-b25a-4bddbfc74ec9.png)

 _App Startup - MainActivity#onCreate is a custom child span added to measure our onCreate implementation_

**Comment by marandaneto** [comment]
@LouisYubo it could be related to opening the App thru notifications, mind testing that out? at least the sample that you have given is fixed, please raise a new issue with more detailed use case for the other one, also a minimal reproducible example would help as it did with this one, thanks a lot.
